### ISSUE TITLE
Web UI not functional behind reverse proxy

### ISSUE THREAD

**Issue by seifertm** [ohwgiles/laminar#6]
When the laminar web UI is behind a reverse proxy (nginx), the website does not work. It does not show any UI elements:
![screenshot](https://user-images.githubusercontent.com/8395347/31428992-053fe47c-ae6d-11e7-92d6-a969e174d22e.png)

The means of connecting to the website does not seem to have any impact. I tried connecting via a unix socket and via a network socket. The nginx rule used to forward requests to laminar is simple:

```
location / {
    #proxy_pass http://unix:/var/lib/laminar/laminard.sock;
    proxy_pass http://example.com:8081;
}
```

When I connect to http://example.com:8081 directly, the web UI is displayed correctly.

**Comment by ohwgiles** [comment]
Laminar uses websockets to transfer all the data the page template uses to render the full UI. By default, nginx does not proxy websocket connections. There's some more information at http://nginx.org/en/docs/http/websocket.html

I just made commit f96e58c which mentions this in the README, but at some point I hope to have a full config snippet for common webservers. Also, I'm not sure that the proxied websocket connections don't time out

**Comment by ohwgiles** [comment]
I was able to (at least partially?) get this working with

```
http {
  map $http_upgrade $connection_upgrade {
    default upgrade;
    '' close;
  }
  server {
    [...]
    location / {
      proxy_pass http://127.0.0.1:8080;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection $connection_upgrade;
    }
  }
}
```

**Comment by seifertm** [comment]
Much appreciated, thanks! After adjusting the nginx config, it worked as soon as I switched to the git version of laminar (the wss:// protocol fix for https is not in the 0.4 release).
